sdk/storage/Azure.Storage.DataMovement/src/TransferJobInternal.cs

```diff
@@ -327,6 +327,7 @@ public async Task JobPartEvent(TransferStatusEventArgs args)
                 jobStatus != StorageTransferStatus.Completed)
             {
                 await TriggerJobCancellationAsync().ConfigureAwait(false);
+                jobStatus = _dataTransfer._state.GetTransferStatus();
```

Comment:
Was this an attempt to fix some flakiness with the StopOnFailure tests? Did you try with my recent fix for this and if so, is this still needed?

Reply:
I think this was a different bug. Cause after we trigger the cancellation, it's possible for the job status to change.

So we need to make sure the jobStatus is also updated so that way it has the current jobStatus when checked by the other if statements below.

Reply:
Hmm that is a good point but I wouldn't think it would have made much of a difference given the below code. The only thing that could happen here (I think) is InProgress -> CancellationInProgress, both of which are treated the same in the below if-statement.

That being said I guess this doesn't hurt. The only thing to consider is that `GetTransferStatus` has a lock check so this is yet another lock we have in this path (which already has a few). I've added to some GitHub issue we have to investigate if we can get rid of some of the locks in the future though.